### DISCUSSION TITLE
Querying ?

### DISCUSSION THREAD

**Discussion by MentalGear** [toeverything/OctoBase#564] [Q&A]
I just found out about this, and it's really impressive what you have build: Kudos!
One thing I didn't find any documentation about is how to do query, and what kind is supported?
.

**Comment by darkskygit** [comment]
Thank you for your interest. Currently you can import this lib and make crdt operations like this:

Cargo.toml:

``` toml
jwst-core = { git = "https://github.com/toeverything/OctoBase", branch = "master" }
```

main.rs:
``` rust
use jwst_core::Workspace;

let workspace = Workspace::new(id);

// do any operations for a workspace
```

Recently, we just completed the rewriting of the underlying crdt implementation and contributed to the official repository of yjs: https://github.com/y-crdt/y-octo, if you wish to perform more in-depth operations, consider importing this library directly.

OctoBase is a database developed based on y-octo. It will provide easier-to-use APIs and features that a database should have, such as data storage, indexes, advanced queries, etc.

We are currently rethinking and designing OctoBase's new APIs based on the experience of rewriting y-octo, so documentation has not yet been provided, but we expect to complete the documentation in the next few months!
